addons/product/models/product.py

```diff
@@ -461,31 +462,37 @@ def open_product_template(self):
                 'res_id': self.product_tmpl_id.id,
                 'target': 'new'}
 
+    def _check_seller(self, seller, partner_id=False, quantity=0.0, date=None, uom_id=False):
+        res = True
+        # Set quantity in UoM of seller
+        quantity_uom_seller = quantity
+        precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+        if quantity_uom_seller and uom_id and uom_id != seller.product_uom:
+            quantity_uom_seller = uom_id._compute_quantity(quantity_uom_seller, seller.product_uom)
+
+        if seller.date_start and seller.date_start > date:
```

Comment:
if date is `None` acording to default param `date=None`, `seller.date_start > date` will crush.

Reply:
@linkeyeasy well its meant to be a private method for select_seller where date is explicitly set, but I suppose everything could be converted to positional arguments if its a big deal, I just didn't want the overhead of the date setting inside the loop and wanted to keep arguments as similar as possible.

Reply:
@linkeyeasy I changed it to positional